scheduler/core/eventsqueue/events.py

```diff
@@ -73,34 +73,21 @@ class WeatherChange(Interruption):
     new_conditions: Conditions
 
 
-@dataclass
-class Blockage(Event, ABC):
-    """
-    Parent class for any interruption that causes a blockage and requires a resume event.
-    """
-    end: Optional[datetime] = None  # needs a resume event
-
-    def ends(self, end: datetime) -> None:
-        self.end = end
-
-    def time_loss(self) -> timedelta:
-        if self.end:
-            return self.end - self.start
-        else:
-            raise ValueError("Can't calculate Blockage time loss without end value.")
-
-
-@dataclass
-class ResumeNight(Interruption):
-    """
-    Event that lets the scheduler knows that the night can be resumed.
-    """
-    ...
-
-
 @final
-class Fault(Blockage):
+@dataclass(unsafe_hash=True)
+class Fault(Interruption):
     """
     Blockage that occurs when one or more resources experience a fault.
     """
+    id: str
+    end: datetime
+    time_loss: timedelta
     affects: FrozenSet[Resource]
+
+
+@final
+@dataclass(unsafe_hash=True)
+class EngTask(Interruption):
+
+    end: datetime
+    time_loss: timedelta
```

Comment:
Yes, I didn't know how to call it so I just went with  `Interruption`

Reply:
`Interruption` works fine for the superclass name, but more what I meant is why do we repeat the fields here instead of moving them up.

If we get rid of `end` and `time_loss` as I propose above, then we can also mark the dataclass as `frozen` and that means we don't need the `unsafe_hash`, and we should try to limit the use of `unsafe_hash` as much as possible since it is inherently unsafe.

At the very least, we should mark mutable fields as not being included in the hash with:

```python
@dataclass(unsafe_hash=True)
class MyClass:
    immutable_field: type
    mutable_field: type = field(hash=False)
```

or something to that extent.